taxjar/taxjar-magento2-extension
# INT-2121: Sales order view transaction sync

### Context

Other TJ integrations have the ability to sync a specific order from UI, but integration would currently require an admin to determine the date that an order was originally placed, then initiate transaction backfill (with force sync enabled) over a date range containing the order's date.

### Description

Adds a "Sync to TaxJar" button to the UI toolbar on the Magento Sales Order view page when an order is complete. Clicking the button will force an API request to sync that individual order and any related credit memos with TaxJar.

### Performance

The addition of this feature should improve user experience while increasing feature parody across TJ connectors.

### Testing

- Manually verified "Sync to TaxJar" button is only present on completed orders
- Unit test coverage of added TJ helper methods
- Unit test coverage of controller classes

1. Enable TJ module and "transaction sync" feature.
2. Create an order
3. View order, observe "Sync to TaxJar" button not displayed, Order not synced.
4. Invoice the order
5. Observe "Sync to TaxJar" button not displayed, Order not synced.
6. Ship the order
7. Observe "Sync to TaxJar" button is now displayed, Order is completed and synced to TaxJar.
8. Click "Sync to TaxJar"
9. Observe sales tax synced at date time is updated to "now", API call can be verified by viewing  `taxjar/transaction.log`

#### Versions
